lib/MiLight/MiLightRadioPL1197_LT8900.cpp

```diff
@@ -0,0 +1,631 @@
+/*
+ * MiLightRadioPL1167_LT89000.cpp
+ *
+ *  Created on: 31 March 2017
+ *      Author: WoodsterDK
+ *
+ *  Very inspired by:
+ *  https://github.com/pmoscetta/authometion-milight/tree/master/Authometion-MiLight
+ *  https://bitbucket.org/robvanderveer/lt8900lib
+ */
+
+#include "MiLightRadioPL1167_LT8900.h"
+#include <SPI.h>
+
+#define DEBUG_PRINTF
+
+/**************************************************************************/
+// Constructor
+/**************************************************************************/
+MiLightRadioPL1167_LT8900::MiLightRadioPL1167_LT8900(byte byCSPin, byte byResetPin, byte byPktFlag, const MiLightRadioConfig& config)
+  :  config(config)
+{
+  _csPin = byCSPin;
+	_pin_pktflag = byPktFlag;
+
+  pinMode(_pin_pktflag, INPUT);
+
+	if (byResetPin > 0)					// If zero then bypass hardware reset
+	{
+		pinMode(byResetPin, OUTPUT);
+		digitalWrite(byResetPin, LOW);
+		delay(200);
+		digitalWrite(byResetPin, HIGH);
+		delay(200);
+	}
+
+  pinMode(_csPin, OUTPUT);
+  digitalWrite(_csPin, HIGH);
+
+  SPI.begin();
+
+  SPI.setFrequency(1000000);
+  SPI.setDataMode(SPI_MODE1);
+  //SPI.setClockDivider(SPI_CLOCK_DIV64); // UNO and 328
+  SPI.setClockDivider(SPI_CLOCK_DIV128); // MEGA and FASTER
+  SPI.setBitOrder(MSBFIRST);
+
+  //Initialize transceiver with correct settings
+  vInitRadioModule(config.type);
+  delay(50);
+
+  // Check if HW is connected
+  bCheckRadioConnection();
+
+  //Reset SPI MODE to default
+  SPI.setDataMode(SPI_MODE0);
+  _waiting = false;
+}
+
+
+
+
+/**************************************************************************/
+// Checks the connection to the radio module by verifying a register setting
+/**************************************************************************/
+bool MiLightRadioPL1167_LT8900::bCheckRadioConnection(void)
+{
+	int iRetValue = 0;
+	uint16_t value_0 = uiReadRegister(0);
+	uint16_t value_1 = uiReadRegister(1);
+
+	if ((value_0 == 0x6fe0) && (value_1 == 0x5681))
+	{
+    #ifdef DEBUG_PRINTF
+		  Serial.println("Radio module running correctly...");
+    #endif
+		iRetValue = 1;
+	}
+	else
+	{
+    #ifdef DEBUG_PRINTF
+		  Serial.println("Failed initializing the radio module...");
+    #endif
+	}
+
+	return iRetValue;
+}
+
+/**************************************************************************/
+// Initialize radio module
+/**************************************************************************/
+void MiLightRadioPL1167_LT8900::vInitRadioModule(MiLightRadioType type)
+{
+	if (type == RGB_CCT)
+	{
+		bool bWriteDefaultDefault = true;  // Is it okay to use the default power up values, without setting them
+
+		regWrite16(0x00, 0x6F, 0xE0, 7);  // Recommended value by PMmicro
```

Comment:
Interesting. I was asking because the 4 or 5 I spot-checked looked like they were the same (e.g., it looks like 0x00 -- 0x07 are the same). Do you know offhand which values are different?